<p>Unused parameters are misleading. Whatever the values passed to such parameters, the behavior will be the same.</p>
<h2>Noncompliant Code Example</h2>
<pre>function doSomething(a : int, b : int) {     // "b" is unused
  compute(a)
}
</pre>
<h2>Compliant Solution</h2>
<pre>function doSomething(a : int) {
  compute(a)
}
</pre>
<h2>Exceptions</h2>
<p>The rule will not raise issues for unused parameters:</p>
<ul>
    <li> in overridden methods</li>
    <li> in non-private methods that only <code>throw</code> or that have empty bodies</li>
    <li> in overridable methods (non-final, or not member of a final class, non-static, non-private), if the parameter
        is documented with a proper
        javadoc.
    </li>
    <li> in annotated methods, unless the annotation is <code>@SuppressWarning("unchecked")</code> or <code>@SuppressWarning("rawtypes")</code>,
        in
        which case the annotation will be ignored (disabled by default)
    </li>
</ul>
<pre>@Override
function doSomething(a : int, b : int) {     // no issue reported on b
  compute(a)
}

public function foo(s : String) {
  // designed to be extended but noop in standard case
}

protected function bar(s : String) {
  //open-closed principle
}

public function qix(s : String) {
  throw new UnsupportedOperationException("This method should be implemented in subclasses")
}

/**
 * @param s This string may be use for further computation in overriding classes
 */
protected function foobar(a : int, s : String) { // no issue, method is overridable and unused parameter has proper javadoc
  compute(a)
}
</pre>
<h2>See</h2>
<ul>
    <li> MISRA C++:2008, 0-1-11 - There shall be no unused parameters (named or unnamed) in nonvirtual functions.</li>
    <li> MISRA C:2012, 2.7 - There should be no unused parameters in functions</li>
    <li><a href="https://www.securecoding.cert.org/confluence/x/NYA5">CERT, MSC12-C.</a> - Detect and remove code that
        has no effect or is never
        executed
    </li>
</ul>
